### ISSUE TITLE
docker rm /redis appears to succeed

### ISSUE THREAD

**Issue by SvenDowideit** [docker/docker#6898]
if you call some commands with a container name prefixed by `/`, the docker daemon fails to error out:

```
docker(remove-the-named-container-before-creating-a-new-one) $ docker rm asdf
Error response from daemon: No such container: asdf
2014/07/08 11:35:57 Error: failed to remove one or more containers
docker(remove-the-named-container-before-creating-a-new-one) $ docker rm /asdf
/asdf
```

and yet in other places, Docker will quietly remove the slash:

```
docker(remove-the-named-container-before-creating-a-new-one) $ docker run -d --name /banana busybox top
9945adccdf3466ec9d7cd3161b997c1d6ec4db6bbe02e28d82ab70d3b78daced
docker(remove-the-named-container-before-creating-a-new-one) $ docker ps
CONTAINER ID        IMAGE                      COMMAND                CREATED             STATUS              PORTS                     NAMES
9945adccdf34        busybox:latest             top                    3 seconds ago       Up 2 seconds                                  banana  
```
There's extra confusion because `docker inspect` will tell you the name contains a `/`

inspired by a docs-review of an old version of #6863

I think we need to make the commands consistent.

**Comment by erjiang** [comment]
I think a similar issue appears with the rename command: #10996

**Comment by brahmaroutu** [comment]
I have changed the code to strip the forward slash from container name before the http request is made and this will allow container names like /foo at cli. I am not sure if this is intended behavior or a bug to fix though.

**Comment by brahmaroutu** [comment]
I will write tests and issue PR if we want to fix this.

**Comment by jfrazelle** [comment]
@brahmaroutu is this also closed by the rename patch you have or a seperate issue

**Comment by brahmaroutu** [comment]
@jfrazelle This is a different issue where docker cli does not work with names that have prefix "/" This applies to all commands ex: docker rm /foo, etc.

**Comment by jfrazelle** [comment]
Ah col thanks for clearing it up

**Comment by runcom** [comment]
closed by #12921
